examples/tv-app/tv-common/clusters/messages/MessagesManager.h

```diff
@@ -22,6 +22,91 @@
 #include <iostream>
 #include <list>
 
+/**
+ * Note on memory management:
+ *
+ * The CachedMessage contains strings and objects
+ * allocated when created (via HandlePresentMessagesRequest) and freed
+ * in the destructor (via HandleCancelMessagesRequest).
+ *
+ */
+
+struct CachedMessage
+{
+    chip::ByteSpan mMessageId;
+
+    const chip::app::Clusters::Messages::MessagePriorityEnum mPriority;
+    const chip::BitMask<chip::app::Clusters::Messages::MessageControlBitmap> mMessageControl;
+    const chip::app::DataModel::Nullable<uint32_t> mStartTime;
+    const chip::app::DataModel::Nullable<uint16_t> mDuration;
+
+    std::string mMessageText;
+
+    chip::app::Clusters::Messages::MessageResponseOption * mOptionArray = nullptr;
+    size_t mOptionArraySize                                             = 0;
+
+    CachedMessage(uint8_t * messageId, size_t messageIdSize, const chip::app::Clusters::Messages::MessagePriorityEnum & priority,
+                  const chip::BitMask<chip::app::Clusters::Messages::MessageControlBitmap> & messageControl,
+                  const chip::app::DataModel::Nullable<uint32_t> & startTime,
+                  const chip::app::DataModel::Nullable<uint16_t> & duration, std::string messageText) :
+        mMessageId(messageId, messageIdSize),
```

Comment:
this style of passing and delete in the destructor separates memory allocation (out of this class) from memory freeing (in this class). This seems not ideal.

Please make memory allocation and management here. One idea would be to have a `ScopedMemoryBuffer` member, do the alloc and memcopy in the constructor. Also have a `Valid` or similar method to check if allocation succeeded on every member. Or alternatively make the constructor private and make this class have a factory method.

You also have to delete the copy and assignment operators to make sure you do not double-free. If you use ScopedMemoryBuffer you should get that for free.